<launch>
    <arg name="range" default="18.0"/>
    <arg name="grid_num" default="80"/>
    <arg name="hz" default="100.0"/>
    <arg name="crop" default="10"/>
    <arg name="is_save_image" default="false"/>
    <arg name="save_data_num" default="2"/>
	<arg name="pkg_path" default="/home/amsl/ros_catkin_ws/src/bev_converter/bev_img"/>

	<param name="use_sim_time" value="true"/>
	
	<node pkg="bev_converter" type="bev_converter" name="bev_converter" output="screen">
		<param name="RANGE" value="$(arg range)" type="double"/>
		<param name="GRID_NUM" value="$(arg grid_num)" type="int"/>
		<param name="Hz" value="$(arg hz)" type="double"/>
	</node>

	<node pkg="bev_converter" type="bev_flow_estimator" name="bev_flow_estimator" output="screen">
		<param name="RANGE" value="$(arg range)" type="double"/>
		<param name="GRID_NUM" value="$(arg grid_num)" type="int"/>
		<param name="Hz" value="$(arg hz)" type="double"/>
		<param name="MANUAL_CROP_SIZE" value="$(arg crop)" type="int"/>
		<param name="IS_SAVE_IMAGE" value="$(arg is_save_image)" type="bool"/>
		<param name="SAVE_NUMBER" value="$(arg save_data_num)" type="int"/>
        <!-- <rosparam command="load" file="$(find cats)/config/cats_motion_param.yaml"/> -->
	</node>

	<!-- <include file="$(find velodyne_pointcloud)/launch/32e_points.launch"/> -->
	<!-- <include file="$(find velodyne_height_map)/launch/amsl_velodyne_heightmap.launch"/> -->
	<include file="$(find cats)/launch/robot_state.launch"/>

	<node pkg="rosbag" name="player" type="play" args="--clock /home/amsl/bagfiles/cats/2019/left_start.bag"/>

	<node pkg="rviz" name="rviz" type="rviz" args="-d /home/amsl/.rviz/bev.rviz"/>
</launch>
